### PR TITLE
feat: add users table 

### PR DESCRIPTION
**Pull request by CristianaPereira**
- added users MC
- handle errors response for portal 
- added tests

### CODE REVIEW

File: app/controllers/api/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class Api::UsersController < ApplicationController
+  before_action :set_user, only: %i[ show update destroy ]
+
+  # GET /users
+  def index
+    @users = User.all
+
+    render json: @users
```
Comment (daniel-carvalho-runtime-revolution) [inline app/controllers/api/users_controller.rb:8]: could be only

`render json: User.all, status: :ok`

File: app/controllers/api/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  # GET /users/1
+  def show
+    render json: @user
```
Comment (daniel-carvalho-runtime-revolution) [inline app/controllers/api/users_controller.rb:13]: missing the `status: :ok` it's not mandatory, but it's a good practice

adding rescue `render json: {error: 'Invalid User'}`, status: :not_found

File: app/controllers/api/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  # POST /users
+  def create
+    @user = User.new(user_params)
```
Comment (daniel-carvalho-runtime-revolution) [inline app/controllers/api/users_controller.rb:18]: `User.create!(user_params).as_json`
render json: @user.as_json(except: [ :password_digest ]), status: :created

rescue ...

File: app/controllers/api/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  def update
+    if @user.update(user_params)
+      render json: @user
+    else
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /users/1
+  def destroy
```
Comment (daniel-carvalho-runtime-revolution) [inline app/controllers/api/users_controller.rb:38]: missing rescue and json response

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:new_user) { User.new(name: "Another User", username: "anotherUser", email: "anotherUser@gmail.com", password: "veryStrongPassword") }
```
Comment (daniel-carvalho-runtime-revolution) [inline spec/models/user_spec.rb:4]: see factory_bot gem

**Review by luispcosta — APPROVED** [review]

**Review by firedeathz — APPROVED** [review]

**Pull request merged**